pyproj/_transformer.pyx

```diff
@@ -712,6 +712,91 @@ cdef class _Transformer(Base):
                     ybuff.data[iii] = ybuff.data[iii]*_DG2RAD
         ProjError.clear()
 
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def _transform_point(
+        self,
+        object inx,
+        object iny,
+        object inz,
+        object intime,
+        object direction,
+        bint radians,
+        bint errcheck,
+    ):
+        """
+        Optimized to transform a single point between two coordinate systems.
+        """
+        cdef double coord_x = inx
+        cdef double coord_y = iny
+        cdef double coord_z = 0
+        cdef double coord_t = HUGE_VAL
```

Comment:
I guess it doesn't matter if these are attempted to cast here before the instance checks below and will raise a TypeError if it can't cast?

Reply:
Yes, that is correct.

Reply:
This the only reason for the isinstance check is for 1D numpy arrays.

Reply:
even more minimal than that... single-value numpy arrays is the only reason :( Definitely seems like a poor decision to allow the array->float cast, but I doubt they would want to change upstream, so we're stuck with this I guess.